### PR TITLE
Improvements to CoreCLR support

### PR DESCRIPTION
**Pull request by jtattermusch**
-- update Google.Apis.Auth to 1.16.0 so that service account credentials are supported on coreCLR
-- upgrade Google.Protobuf to 3.0.0
-- upgrade tests to newest version of Moq (which supports CoreCLR), and enable a related test under coreclr.
-- remove unnecessary imports from project.json projects (and other cleanup).

Not all the "portable-net45" import could be removed because they are required by NUnit (that does not properly support netstandard1.X yet)

### CODE REVIEW

**Comment by jtattermusch** [comment]
C# tests seem to be green enough, so this should be ready for review. In the meantime, the non C# related problems with portability tests and interop tests will hopefully get solved.

**Comment by apolcyn** [comment]
I think distrib test files need updates too?
https://github.com/grpc/grpc/blob/v1.0.x/test/distrib/csharp/DistribTest/DistribTest.csproj
https://github.com/grpc/grpc/blob/v1.0.x/test/distrib/csharp/DistribTest/packages.config

File: src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj [hunk 1]
```diff
@@ -63,17 +57,38 @@
...
+    </Reference>
+    <Reference Include="log4net">
+      <HintPath>..\packages\log4net.2.0.3\lib\net40-full\log4net.dll</HintPath>
+    </Reference>
     <Reference Include="Google.Apis.Core">
-      <HintPath>..\packages\Google.Apis.Core.1.15.0\lib\net45\Google.Apis.Core.dll</HintPath>
+      <HintPath>..\packages\Google.Apis.Core.1.16.0\lib\net45\Google.Apis.Core.dll</HintPath>
+    </Reference>
+    <Reference Include="Zlib.Portable">
+      <HintPath>..\packages\Zlib.Portable.Signed.1.11.0\lib\portable-net4+sl5+wp8+win8+wpa81+MonoTouch+MonoAndroid\Zlib.Portable.dll</HintPath>
```
Comment (apolcyn) [inline src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj]: Was more just wondering on the addition of this one. A package rename/split? With "net4", the target name seems odd too, but I guess this has more to just to do with that package impl.

File: src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj [hunk 2]
```diff
@@ -63,17 +57,38 @@
     <Reference Include="nunitlite">
       <HintPath>..\packages\NUnitLite.3.2.0\lib\net45\nunitlite.dll</HintPath>
     </Reference>
+    <Reference Include="CommandLineParser.Unofficial">
+      <HintPath>..\packages\CommandLineParser.Unofficial.2.0.275\lib\net45\CommandLineParser.Unofficial.dll</HintPath>
+    </Reference>
+    <Reference Include="log4net">
+      <HintPath>..\packages\log4net.2.0.3\lib\net40-full\log4net.dll</HintPath>
```
Comment (apolcyn) [inline src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj]: same thoughts as with the Zlib reference

**Comment by jtattermusch** [comment]
I didn't initially update distribtests because they were out of sync with v1.0.x branch so I wanted to prevent conflicts. I will either update them here or in a separate PR.

File: src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj (hunk 1 above)
Comment (jtattermusch) [inline src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj]: This is a new dependency of Google.Apis.*. The older versions of nuget (pre dotnet CLI) are a bit stupid and they unpack all the transitive dependencies to the project. So it looks like I added Zlib myself, but instead the IDE adds it for you. The line is correct.

File: src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj (hunk 2 above)
Comment (jtattermusch) [inline src/csharp/Grpc.IntegrationTesting/Grpc.IntegrationTesting.csproj]: Same as above.

**Comment by jtattermusch** [comment]
Tests seem reasonably green.

**Comment by apolcyn** [comment]
LGTM

**Pull request merged**